From 0932392f6f61c7c6ab80d1e8e51ac2e0e3ca0f71 Mon Sep 17 00:00:00 2001
From: Patryk Kups <pkups@anaconda.com>
Date: Fri, 16 Jan 2026 18:22:45 +0100
Subject: [PATCH] patch

---
 test/unit/psset.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test/unit/psset.c b/test/unit/psset.c
index 6ff7201..c8d2e77 100644
--- a/test/unit/psset.c
+++ b/test/unit/psset.c
@@ -1,7 +1,10 @@
+#define HUGEPAGE_PAGES 1
 #include "test/jemalloc_test.h"
 
 #include "jemalloc/internal/psset.h"
 
+
+
 #define PAGESLAB_ADDR ((void *)(1234 * HUGEPAGE))
 #define PAGESLAB_AGE 5678
 
-- 
2.39.5 (Apple Git-154)

