From d1a409c4c84d56c98ecbf95686ac8c051e37abc3 Mon Sep 17 00:00:00 2001
From: Patryk Kups <pkups@anaconda.com>
Date: Fri, 16 Jan 2026 11:46:04 +0100
Subject: [PATCH] patch

---
 test/unit/psset.c | 49 +++++++++++++++++++++++++++++++++++------------
 1 file changed, 37 insertions(+), 12 deletions(-)

diff --git a/test/unit/psset.c b/test/unit/psset.c
index 6ff7201..d46290d 100644
--- a/test/unit/psset.c
+++ b/test/unit/psset.c
@@ -733,16 +733,41 @@ TEST_END
 
 int
 main(void) {
-	return test_no_reentrancy(
-	    test_empty,
-	    test_fill,
-	    test_reuse,
-	    test_evict,
-	    test_multi_pageslab,
-	    test_stats,
-	    test_oldest_fit,
-	    test_insert_remove,
-	    test_purge_prefers_nonhuge,
-	    test_purge_prefers_empty,
-	    test_purge_prefers_empty_huge);
+	test_status_t status = test_no_reentrancy(test_reuse);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_evict);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_multi_pageslab);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_stats);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_oldest_fit);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_insert_remove);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_purge_prefers_nonhuge);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_purge_prefers_empty);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_purge_prefers_empty_huge);
+	if (status != test_status_pass) {
+		return status;
+	}
+	return test_status_pass;
 }
-- 
2.39.5 (Apple Git-154)

