From 1e3ee166cf905d3bf665c2682f2c77d9e6263fa0 Mon Sep 17 00:00:00 2001
From: Patryk Kups <pkups@anaconda.com>
Date: Thu, 15 Jan 2026 19:09:51 +0100
Subject: [PATCH] patch

---
 Makefile.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Makefile.in b/Makefile.in
index 1193cd8..e810951 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -439,6 +439,13 @@ $(C_JET_SYM_OBJS): CPPFLAGS += -DJEMALLOC_JET -DJEMALLOC_NO_PRIVATE_NAMESPACE
 $(C_JET_SYMS): $(objroot)src/%.jet.sym: $(objroot)src/%.jet.sym.$(O)
 $(C_JET_OBJS): $(objroot)src/%.jet.$(O): $(srcroot)src/%.c
 $(C_JET_OBJS): CPPFLAGS += -DJEMALLOC_JET
+
+TEST_UNIT_CFLAGS := $(filter-out -O0 -O1 -O2 -O3 -Ofast -Og,$(CFLAGS)) \
+                    -O0 -g -fno-builtin -fno-strict-aliasing -fno-tree-sra \
+                    -fno-reorder-blocks -fno-inline -fno-omit-frame-pointer
+
+$(C_TESTLIB_UNIT_OBJS): CFLAGS := $(TEST_UNIT_CFLAGS)
+$(TESTS_UNIT_OBJS): CFLAGS := $(TEST_UNIT_CFLAGS)
 $(C_TESTLIB_UNIT_OBJS): $(objroot)test/src/%.unit.$(O): $(srcroot)test/src/%.c
 $(C_TESTLIB_UNIT_OBJS): CPPFLAGS += -DJEMALLOC_UNIT_TEST
 $(C_TESTLIB_INTEGRATION_OBJS): $(objroot)test/src/%.integration.$(O): $(srcroot)test/src/%.c
-- 
2.39.5 (Apple Git-154)

