From 097828c667bb047d06fde5191497cbda75ca9a78 Mon Sep 17 00:00:00 2001
From: Patryk Kups <pkups@anaconda.com>
Date: Fri, 16 Jan 2026 12:48:30 +0100
Subject: [PATCH] patch

---
 test/unit/psset.c | 117 ++++++++++++++++++++++++++++++----------------
 1 file changed, 78 insertions(+), 39 deletions(-)

diff --git a/test/unit/psset.c b/test/unit/psset.c
index 6ff7201..8c8ed43 100644
--- a/test/unit/psset.c
+++ b/test/unit/psset.c
@@ -1,3 +1,4 @@
+#include <stdio.h>
 #include "test/jemalloc_test.h"
 
 #include "jemalloc/internal/psset.h"
@@ -35,16 +36,16 @@ test_psset_alloc_new(psset_t *psset, hpdata_t *ps, edata_t *r_edata,
 
 	test_psset_fake_purge(ps);
 
-	psset_insert(psset, ps);
-	psset_update_begin(psset, ps);
+	// psset_insert(psset, ps);
+	// psset_update_begin(psset, ps);
 
-        void *addr = hpdata_reserve_alloc(ps, size);
-        edata_init(r_edata, edata_arena_ind_get(r_edata), addr, size,
-	    /* slab */ false, SC_NSIZES, /* sn */ 0, extent_state_active,
-            /* zeroed */ false, /* committed */ true, EXTENT_PAI_HPA,
-            EXTENT_NOT_HEAD);
-        edata_ps_set(r_edata, ps);
-	psset_update_end(psset, ps);
+    //     void *addr = hpdata_reserve_alloc(ps, size);
+    //     edata_init(r_edata, edata_arena_ind_get(r_edata), addr, size,
+	//     /* slab */ false, SC_NSIZES, /* sn */ 0, extent_state_active,
+    //         /* zeroed */ false, /* committed */ true, EXTENT_PAI_HPA,
+    //         EXTENT_NOT_HEAD);
+    //     edata_ps_set(r_edata, ps);
+	// psset_update_end(psset, ps);
 }
 
 static bool
@@ -115,6 +116,8 @@ TEST_BEGIN(test_empty) {
 	/* Empty psset should return fail allocations. */
 	err = test_psset_alloc_reuse(&psset, &alloc, PAGE);
 	expect_true(err, "Empty psset succeeded in an allocation.");
+	fprintf(stderr, "test1\n");
+	fflush(stderr);
 }
 TEST_END
 
@@ -125,28 +128,31 @@ TEST_BEGIN(test_fill) {
 	hpdata_init(&pageslab, PAGESLAB_ADDR, PAGESLAB_AGE);
 
 	edata_t alloc[HUGEPAGE_PAGES];
+	edata_init_test(&alloc[0]);
 
 	psset_t psset;
 	psset_init(&psset);
-
-	edata_init_test(&alloc[0]);
+	fprintf(stderr, "test1\n");
+	fflush(stderr);
 	test_psset_alloc_new(&psset, &pageslab, &alloc[0], PAGE);
-	for (size_t i = 1; i < HUGEPAGE_PAGES; i++) {
-		edata_init_test(&alloc[i]);
-		err = test_psset_alloc_reuse(&psset, &alloc[i], PAGE);
-		expect_false(err, "Nonempty psset failed page allocation.");
-	}
-
-	for (size_t i = 0; i < HUGEPAGE_PAGES; i++) {
-		edata_t *edata = &alloc[i];
-		edata_expect(edata, i, 1);
-	}
-
-	/* The pageslab, and thus psset, should now have no allocations. */
-	edata_t extra_alloc;
-	edata_init_test(&extra_alloc);
-	err = test_psset_alloc_reuse(&psset, &extra_alloc, PAGE);
-	expect_true(err, "Alloc succeeded even though psset should be empty");
+	fprintf(stderr, "test2\n");
+	fflush(stderr);
+	// for (size_t i = 1; i < HUGEPAGE_PAGES; i++) {
+	// 	edata_init_test(&alloc[i]);
+	// 	err = test_psset_alloc_reuse(&psset, &alloc[i], PAGE);
+	// 	expect_false(err, "Nonempty psset failed page allocation.");
+	// }
+
+	// for (size_t i = 0; i < HUGEPAGE_PAGES; i++) {
+	// 	edata_t *edata = &alloc[i];
+	// 	edata_expect(edata, i, 1);
+	// }
+
+	// /* The pageslab, and thus psset, should now have no allocations. */
+	// edata_t extra_alloc;
+	// edata_init_test(&extra_alloc);
+	// err = test_psset_alloc_reuse(&psset, &extra_alloc, PAGE);
+	// expect_true(err, "Alloc succeeded even though psset should be empty");
 }
 TEST_END
 
@@ -733,16 +739,49 @@ TEST_END
 
 int
 main(void) {
-	return test_no_reentrancy(
-	    test_empty,
-	    test_fill,
-	    test_reuse,
-	    test_evict,
-	    test_multi_pageslab,
-	    test_stats,
-	    test_oldest_fit,
-	    test_insert_remove,
-	    test_purge_prefers_nonhuge,
-	    test_purge_prefers_empty,
-	    test_purge_prefers_empty_huge);
+	test_status_t status = test_no_reentrancy(test_empty);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_fill);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_reuse);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_evict);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_multi_pageslab);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_stats);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_oldest_fit);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_insert_remove);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_purge_prefers_nonhuge);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_purge_prefers_empty);
+	if (status != test_status_pass) {
+		return status;
+	}
+	status = test_no_reentrancy(test_purge_prefers_empty_huge);
+	if (status != test_status_pass) {
+		return status;
+	}
+	return test_status_pass;
 }
-- 
2.39.5 (Apple Git-154)

